open-widget-framework/src/defaults.js

```diff
@@ -20,34 +20,69 @@ class _defaultRenderer extends Component {
 }
 
 class _defaultListWrapper extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      editMode: false,
+      addWidgetForm: false,
+      editWidgetForm: null,
+    }
+    this.renderAddWidgetButton = this.renderAddWidgetButton.bind(this)
+    this.closeForm = this.closeForm.bind(this)
+  }
+
   renderAddWidgetButton() {
     return (
-      <button className={'btn btn-info'} onClick={this.props.openNewWidgetForm}>
+      <button className={'btn btn-info'} onClick={() => this.setState({
+        addWidgetForm: !this.state.addWidgetForm,
+        editWidgetForm: null,
+      })}>
         <Octicon name={'plus'}/>
       </button>
     )
   }
+
+  closeForm() {
+    this.setState({
+      addWidgetForm: false,
+      editWidgetForm: null,
+    })
+  }
+
   render() {
     return (
-      <div>
-        {this.props.renderWidgetForm()}
+      <div className={'bg-secondary rounded float-right col-lg-4'}>
+        {this.state.addWidgetForm ? this.props.renderNewWidgetForm({closeForm: this.closeForm}) : null}
+        {this.state.editWidgetForm !== null
+          ? this.props.renderEditWidgetForm(this.state.editWidgetForm, {closeForm: this.closeForm}) : null}
         <div className={'edit-widget-list-bar btn-group'} role={'group'}>
```

Comment:
```suggestion
        <div className="edit-widget-list-bar btn-group" role="group">
```
String properties don't need brackets, can you fix this and other instances too?

Reply:
Yeah my IDE adds them automatincally. I'll fix it.